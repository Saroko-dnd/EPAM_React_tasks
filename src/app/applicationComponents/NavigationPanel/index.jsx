import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

import createUuidv4 from '../../helpers/createUuidv4';

const NavigationPanel = props => (
    <nav
        className={`navigation-panel nav nav-pills d-flex justify-content-around ${
            props.className
        }`}
    >
        {props.navigationLinks.map(navLinkInfo => (
            <NavLink
                key={createUuidv4()}
                className="nav-item nav-link"
                to={navLinkInfo.destination}
                exact
            >
                {navLinkInfo.title}
            </NavLink>
        ))}
    </nav>
);

NavigationPanel.propTypes = {
    navigationLinks: PropTypes.arrayOf(PropTypes.object).isRequired,
    className: PropTypes.string,
};

NavigationPanel.defaultProps = {
    className: '',
};

export default NavigationPanel;
