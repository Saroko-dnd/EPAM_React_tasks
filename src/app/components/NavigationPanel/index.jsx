import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

const NavigationPanel = ({ navLinksCallback, navigationLinks, className }) => (
    <nav
        className={`navigation-panel nav nav-pills justify-content-around ${className}`}
    >
        {navigationLinks.map(navLinkInfo => (
            <NavLink
                onClick={() => {
                    navLinksCallback();
                }}
                key={navLinkInfo.id}
                className="nav-item nav-link"
                to={navLinkInfo.destination}
                isActive={navLinkInfo.isActive}
                exact
            >
                {navLinkInfo.title}
            </NavLink>
        ))}
    </nav>
);

NavigationPanel.propTypes = {
    navLinksCallback: PropTypes.func.isRequired,
    navigationLinks: PropTypes.arrayOf(PropTypes.object).isRequired,
    className: PropTypes.string,
};

NavigationPanel.defaultProps = {
    className: '',
};

export default NavigationPanel;
