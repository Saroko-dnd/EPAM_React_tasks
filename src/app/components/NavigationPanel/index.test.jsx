import ReactRouterEnzymeContext from 'react-router-enzyme-context';

import NavigationPanel from '.';

describe('render a NavigationPanel', () => {
    const options = new ReactRouterEnzymeContext();
    const clickEventHandler = jest.fn();
    const navLinks = [
        {
            id: 1,
            destination: 'destination 1',
        },
        {
            id: 2,
            destination: 'destination 2',
        },
    ];
    const navigationPanel = shallow(
        <NavigationPanel
            navLinksCallback={clickEventHandler}
            navigationLinks={navLinks}
        />,
        options.get(),
    );

    it('should render NavigationPanel correctly', () => {
        expect(navigationPanel).toMatchSnapshot();
    });

    it('should call function passed as navLinksCallback if one of the links was clicked', () => {
        navigationPanel
            .find('.nav-item.nav-link')
            .at(0)
            .simulate('click');
        expect(clickEventHandler).toHaveBeenCalledTimes(1);

        navigationPanel
            .find('.nav-item.nav-link')
            .at(1)
            .simulate('click');
        expect(clickEventHandler).toHaveBeenCalledTimes(2);
    });
});
