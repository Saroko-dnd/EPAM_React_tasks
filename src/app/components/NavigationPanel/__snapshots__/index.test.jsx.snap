// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a NavigationPanel should render NavigationPanel correctly 1`] = `
<nav
  className="navigation-panel flex-column flex-lg-row align-self-stretch nav nav-pills justify-content-center align-items-center "
>
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="nav-item nav-link"
    exact={true}
    key="1"
    onClick={[Function]}
    to="destination 1"
  />
  <NavLink
    activeClassName="active"
    aria-current="page"
    className="nav-item nav-link"
    exact={true}
    key="2"
    onClick={[Function]}
    to="destination 2"
  />
</nav>
`;
